internal/server/version.go

```diff
@@ -0,0 +1,58 @@
+package server
+
+import (
+	"strconv"
+	"strings"
+)
+
+type Version struct {
+	Major uint8
+	Minor uint8
+	Patch uint8
+}
+
+func NewVersion(ver string) Version {
```

Comment:
Could we use a non-semantic version to make this easier to parse? We only need migrations for breaking changes anyway, so semantic versions don't seem like they would apply here. Maybe just an sequential int, or some form of encoded date (`20220405`) ?

Reply:
we only publish semantic versions, so the client can get no better resolution than that.